ocs_ci/helpers/managed_services.py

```diff
@@ -263,13 +265,20 @@ def verify_storageclient(
         else storageclient_obj.get()["items"][0]
     )
     storageclient_name = storageclient["metadata"]["name"]
-    provider_name = provider_name or config.ENV_DATA.get("provider_name", "")
-    endpoint_actual = get_storage_provider_endpoint(provider_name)
-    assert storageclient["spec"]["storageProviderEndpoint"] == endpoint_actual, (
-        f"The value of storageProviderEndpoint is not correct in the storageclient {storageclient['metadata']['name']}."
-        f" Value in storageclient is {storageclient['spec']['storageProviderEndpoint']}. "
-        f"Value in the provider cluster {provider_name} is {endpoint_actual}"
-    )
+
+    if ocs_version >= version.VERSION_4_16:
+        assert (
+            storageclient["spec"]["storageProviderEndpoint"]
+            == "ocs-provider-server:50051"
+        )
+    else:
+        provider_name = provider_name or config.ENV_DATA.get("provider_name", "")
+        endpoint_actual = get_storage_provider_endpoint(provider_name)
```

Comment:
the assertion bellow will fail, the function works only comparing the endpoint to the one that we put to configuration `config.DEPLOYMENT.get("storage_provider_endpoint", "")`
afaik we don't provide it. Please correct me if we do. 
The provider_name name is used in this function, the function is not finished:

```
def get_storage_provider_endpoint(cluster):
    """
    Get get_storage_provider_endpoint

    Args:
        cluster (str): cluster name

    Returns:
        str: value of storage provider endpoint

    """

    # TODO: p2 task to implement below functionality
    #  Use multicluster implementation to use
    #  kubeconfig as per cluster name and
    #  extract value of storage_provider_endpoint
    #  handle invalid cluster name in implementation
    #  validate Return String storage provider endpoint:
    #     1. raise Error if storage_provider_endpoint is
    #        not found in cluster yaml
    #     2. warning if storage cluster is not ready
    #     and storage_provider_endpoint is available in
    #     storagecluster yaml .
    #  For now use hardcoded value from config with key
    #  storage_provider_endpoint:
    return config.DEPLOYMENT.get("storage_provider_endpoint", "")
```

Reply:
Have added method, fetch_provider_endpoint() under ocs_ci.ocs.resources.storage_client.py